{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import re\n",
    "from nltk.corpus import stopwords \n",
    "from nltk.stem.porter import PorterStemmer "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer, TfidfTransformer\n",
    "from sklearn.grid_search import GridSearchCV \n",
    "from sklearn.metrics import accuracy_score\n",
    "from sklearn import svm\n",
    "from sklearn import preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import xgboost as xgb\n",
    "from xgboost.sklearn import XGBClassifier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "sns.plotting_context('poster')\n",
    "sns.set_style('whitegrid')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### training datasets\n",
    "variants and text files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_v = pd.read_csv('training_variants/training_variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_t = pd.read_csv('training_text/training_text', sep='\\|\\|',skiprows=1, engine='python', names=[\"ID\",\"text\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train = pd.merge(train_v,train_t, on = ['ID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train['data'] = 'train'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### testing datasets\n",
    "variant and text files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test_v = pd.read_csv('test_variants/test_variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test_t = pd.read_csv('test_text/test_text', sep='\\|\\|',skiprows=1, engine='python', names=[\"ID\",\"text\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test = pd.merge(test_v,test_t, on = ['ID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test['data'] = 'test'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### combine training and testing datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df = pd.merge(train, test, how='outer')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Features"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variants\n",
    "Classifies variations into common types and standardizes their name. The order of commands in the fuc matters because deletions & insertions can happen simulteneously but coding is inconsistent. Testing: func is coding variations accurately. We might want to change the Fussion expression (\"fs\") because it could be problematic with new data, but hacky version works for now."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def var_recode(data, colname):\n",
    "    recoded_colname = colname + '_recoded'\n",
    "    #recode substitutions\n",
    "    data[recoded_colname] = data[colname].str.replace('^[A-Z]\\d+[A-Z\\*]$|^(null)\\d+[A-Z\\*]$', 'Substitution')\n",
    "    #recode deletions\n",
    "    data.loc[data[colname].str.contains('del|silencing|hypermethylation', case=False), recoded_colname] = 'Deletion'\n",
    "    #recode insertions\n",
    "    data.loc[data[colname].str.contains('ins', case=False) , recoded_colname] = 'Insertion'\n",
    "    #recode deletions/insertions\n",
    "    data.loc[(data[colname].str.contains('del', case=False) & \n",
    "              data[colname].str.contains('ins', case=False)), recoded_colname] = 'InDel'\n",
    "    #recode truncations\n",
    "    data.loc[data[colname].str.contains('trunc', case=False), recoded_colname] = 'Truncation'\n",
    "    #recode duplications\n",
    "    data.loc[data[colname].str.contains('dup', case=False) , recoded_colname] = 'Duplication'\n",
    "    #recode fusions \n",
    "    data.loc[data[colname].str.contains('fusion|fs', case=False), recoded_colname] = 'Fusion'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "var_recode(df, 'Variation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>text</th>\n",
       "      <th>data</th>\n",
       "      <th>Variation_recoded</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>train</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>train</td>\n",
       "      <td>Substitution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>train</td>\n",
       "      <td>Substitution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>train</td>\n",
       "      <td>Substitution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>train</td>\n",
       "      <td>Substitution</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene             Variation  Class  \\\n",
       "0   0  FAM58A  Truncating Mutations    1.0   \n",
       "1   1     CBL                 W802*    2.0   \n",
       "2   2     CBL                 Q249E    2.0   \n",
       "3   3     CBL                 N454D    3.0   \n",
       "4   4     CBL                 L399V    4.0   \n",
       "\n",
       "                                                text   data Variation_recoded  \n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...  train        Truncation  \n",
       "1   Abstract Background  Non-small cell lung canc...  train      Substitution  \n",
       "2   Abstract Background  Non-small cell lung canc...  train      Substitution  \n",
       "3  Recent evidence has demonstrated that acquired...  train      Substitution  \n",
       "4  Oncogenic mutations in the monomeric Casitas B...  train      Substitution  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Classify the type of substitution: neg to pos, pos to neg, neg to neutral, etc. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#aminoacid: charge\n",
    "sub_dict= {'A': 'Neu', #'alanine'\n",
    "           'R': 'Pos', #'arginine'\n",
    "           'N': 'Neu', #'asparagine'\n",
    "           'D': 'Neg', #'aspartic acid'\n",
    "           'B': 'Neu', #'asparagine|aspartic acid'\n",
    "           'C': 'Neu', #'cysteine'\n",
    "           'E': 'Neg', #'glutamic acid'\n",
    "           'Q': 'Neu', #'glutamine'\n",
    "           'Z': 'Neu', #'glutamine|glutamic acid'\n",
    "           'G': 'Neu', #'glycine'\n",
    "           'H': 'Pos', #'histidine'\n",
    "           'I': 'Neu', #'isoleucine'\n",
    "           'L': 'Neu', #'leucine'\n",
    "           'K': 'Pos', #'lysine'\n",
    "           'M': 'Neu', #'methionine'\n",
    "           'F': 'Neu', #'phenylalanine'\n",
    "           'P': 'Neu', #'proline'\n",
    "           'S': 'Neu', #'serine'\n",
    "           'T': 'Neu', #'threonine'\n",
    "           'W': 'Neu', #'tryptophan'\n",
    "           'Y': 'Neu', #'tyrosine'\n",
    "           'V': 'Neu' #'valine'\n",
    "          }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def sub_recode(data, colname):\n",
    "    recoded_colname = colname + '_recoded'\n",
    "    data['s1'] = data.loc[data[colname].str.match('^[A-Z]\\d+[A-Z]$')][colname].str[:1]\n",
    "    data['s2'] = data.loc[data[colname].str.match('^[A-Z]\\d+[A-Z]$')][colname].str[-1:]\n",
    "    data['s1'].replace(sub_dict, inplace=True)\n",
    "    data['s2'].replace(sub_dict, inplace=True)\n",
    "    data['Sub_type'] = data['s1'] + data['s2']\n",
    "    data.loc[data['Sub_type'].notnull(), recoded_colname] = data['Sub_type']\n",
    "    data.drop(['s1', 's2', 'Sub_type'], axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sub_recode(df, 'Variation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>text</th>\n",
       "      <th>data</th>\n",
       "      <th>Variation_recoded</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>train</td>\n",
       "      <td>Truncation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>train</td>\n",
       "      <td>Substitution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>train</td>\n",
       "      <td>NeuNeg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>train</td>\n",
       "      <td>NeuNeg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>train</td>\n",
       "      <td>NeuNeu</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene             Variation  Class  \\\n",
       "0   0  FAM58A  Truncating Mutations    1.0   \n",
       "1   1     CBL                 W802*    2.0   \n",
       "2   2     CBL                 Q249E    2.0   \n",
       "3   3     CBL                 N454D    3.0   \n",
       "4   4     CBL                 L399V    4.0   \n",
       "\n",
       "                                                text   data Variation_recoded  \n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...  train        Truncation  \n",
       "1   Abstract Background  Non-small cell lung canc...  train      Substitution  \n",
       "2   Abstract Background  Non-small cell lung canc...  train            NeuNeg  \n",
       "3  Recent evidence has demonstrated that acquired...  train            NeuNeg  \n",
       "4  Oncogenic mutations in the monomeric Casitas B...  train            NeuNeu  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## genes\n",
    "quick function to calculate gene frequency for a grouping variable 1) count of unique genes 2) total number of genes per group (similar to group frequency) 3) most frequent gene in each group 4) proportional frequency of gene accounting for size of group 5) reset index for easier plotting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def genefreq(data, group):\n",
    "    newdf = data.groupby(group).agg({'Gene': pd.Series.nunique})\n",
    "    newdf['Gene_total'] = data.groupby(group)['Gene'].count()\n",
    "    newdf['Gene_most_frequent'] = data.groupby(group)['Gene'].agg(lambda x: x.value_counts().idxmax())\n",
    "    newdf['Gene_most_frequent_count'] = data.groupby(group)['Gene'].agg(lambda x: x.value_counts().max())\n",
    "    newdf['Gene_p_unique'] = newdf['Gene']/newdf['Gene_total']\n",
    "    newdf['Gene_p_most_frequent'] = newdf['Gene_most_frequent_count']/newdf['Gene_total']\n",
    "    newdf.reset_index(inplace=True)\n",
    "    return newdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation_recoded</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_total</th>\n",
       "      <th>Gene_most_frequent</th>\n",
       "      <th>Gene_most_frequent_count</th>\n",
       "      <th>Gene_p_unique</th>\n",
       "      <th>Gene_p_most_frequent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>596_619splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HLA-A</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>963_D1010splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MET</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>981_1028splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MET</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A113_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EIF1AX</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AR-V7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>AR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ARv567es</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>AR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Amplification</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>KMT2A</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.012658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>C1385</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EP300</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CASP8L</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>CASP8</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Copy Number Loss</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>FBXW7</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>DNA binding domain missense mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DNMT3B7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>111</td>\n",
       "      <td>199</td>\n",
       "      <td>KIT</td>\n",
       "      <td>34</td>\n",
       "      <td>0.557789</td>\n",
       "      <td>0.170854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>6</td>\n",
       "      <td>0.562500</td>\n",
       "      <td>0.375000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>EGFR-KDD</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>EGFRvII</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>EGFRvIII</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>EGFRvIV</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>EGFRvV</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Exon 1 mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MED12</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Exon 11 mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KIT</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Exon 2 mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MED12</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Exon 9 mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KIT</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Fusion</td>\n",
       "      <td>57</td>\n",
       "      <td>219</td>\n",
       "      <td>PDGFRB</td>\n",
       "      <td>19</td>\n",
       "      <td>0.260274</td>\n",
       "      <td>0.086758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>InDel</td>\n",
       "      <td>9</td>\n",
       "      <td>39</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>15</td>\n",
       "      <td>0.230769</td>\n",
       "      <td>0.384615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>11</td>\n",
       "      <td>50</td>\n",
       "      <td>FLT3</td>\n",
       "      <td>15</td>\n",
       "      <td>0.220000</td>\n",
       "      <td>0.300000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>L225LI</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>L232LI</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>CARD11</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MYC-nick</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MYC</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NegNeg</td>\n",
       "      <td>56</td>\n",
       "      <td>60</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.933333</td>\n",
       "      <td>0.050000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NegNeu</td>\n",
       "      <td>263</td>\n",
       "      <td>517</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>24</td>\n",
       "      <td>0.508704</td>\n",
       "      <td>0.046422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NegPos</td>\n",
       "      <td>180</td>\n",
       "      <td>284</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>11</td>\n",
       "      <td>0.633803</td>\n",
       "      <td>0.038732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NeuNeg</td>\n",
       "      <td>294</td>\n",
       "      <td>458</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>12</td>\n",
       "      <td>0.641921</td>\n",
       "      <td>0.026201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NeuNeu</td>\n",
       "      <td>1065</td>\n",
       "      <td>4002</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>157</td>\n",
       "      <td>0.266117</td>\n",
       "      <td>0.039230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NeuPos</td>\n",
       "      <td>540</td>\n",
       "      <td>976</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>41</td>\n",
       "      <td>0.553279</td>\n",
       "      <td>0.042008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Overexpression</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>CCNE1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.166667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>PosNeg</td>\n",
       "      <td>74</td>\n",
       "      <td>90</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>5</td>\n",
       "      <td>0.822222</td>\n",
       "      <td>0.055556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>PosNeu</td>\n",
       "      <td>630</td>\n",
       "      <td>1393</td>\n",
       "      <td>TP53</td>\n",
       "      <td>50</td>\n",
       "      <td>0.452261</td>\n",
       "      <td>0.035894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PosPos</td>\n",
       "      <td>255</td>\n",
       "      <td>369</td>\n",
       "      <td>TP53</td>\n",
       "      <td>15</td>\n",
       "      <td>0.691057</td>\n",
       "      <td>0.040650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Promoter Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>TERT</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>R1627</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>EP300</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Single Nucleotide Polymorphism</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Substitution</td>\n",
       "      <td>47</td>\n",
       "      <td>76</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>12</td>\n",
       "      <td>0.618421</td>\n",
       "      <td>0.157895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>TGFBR1*6A</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>TGFBR1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>TRKAIII Splice Variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Truncation</td>\n",
       "      <td>115</td>\n",
       "      <td>118</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.974576</td>\n",
       "      <td>0.016949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Wildtype</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>X1006_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MET</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>X1007_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MET</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>X1008_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MET</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>X1009_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MET</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>X434_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>X475_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>X582_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>X963_splice</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>MET</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>p61BRAF</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Variation_recoded  Gene  Gene_total  \\\n",
       "0                           596_619splice     1           1   \n",
       "1                         963_D1010splice     1           1   \n",
       "2                          981_1028splice     1           1   \n",
       "3                             A113_splice     1           1   \n",
       "4                                   AR-V7     1           1   \n",
       "5                                ARv567es     1           1   \n",
       "6                           Amplification    79          79   \n",
       "7                                   C1385     1           1   \n",
       "8                                  CASP8L     1           1   \n",
       "9                        Copy Number Loss     1           1   \n",
       "10  DNA binding domain missense mutations     1           1   \n",
       "11                                DNMT3B7     1           1   \n",
       "12                               Deletion   111         199   \n",
       "13                            Duplication     9          16   \n",
       "14                               EGFR-KDD     1           1   \n",
       "15                                EGFRvII     1           1   \n",
       "16                               EGFRvIII     1           1   \n",
       "17                                EGFRvIV     1           1   \n",
       "18                                 EGFRvV     1           1   \n",
       "19                       Exon 1 mutations     1           1   \n",
       "20                      Exon 11 mutations     1           1   \n",
       "21                       Exon 2 mutations     1           1   \n",
       "22                       Exon 9 mutations     1           1   \n",
       "23                                 Fusion    57         219   \n",
       "24                                  InDel     9          39   \n",
       "25                              Insertion    11          50   \n",
       "26                                 L225LI     1           1   \n",
       "27                                 L232LI     1           1   \n",
       "28                               MYC-nick     1           1   \n",
       "29                                 NegNeg    56          60   \n",
       "30                                 NegNeu   263         517   \n",
       "31                                 NegPos   180         284   \n",
       "32                                 NeuNeg   294         458   \n",
       "33                                 NeuNeu  1065        4002   \n",
       "34                                 NeuPos   540         976   \n",
       "35                         Overexpression     6           6   \n",
       "36                                 PosNeg    74          90   \n",
       "37                                 PosNeu   630        1393   \n",
       "38                                 PosPos   255         369   \n",
       "39                     Promoter Mutations     1           1   \n",
       "40                                  R1627     1           1   \n",
       "41         Single Nucleotide Polymorphism     1           1   \n",
       "42                           Substitution    47          76   \n",
       "43                              TGFBR1*6A     1           1   \n",
       "44                 TRKAIII Splice Variant     1           1   \n",
       "45                             Truncation   115         118   \n",
       "46                               Wildtype     1           1   \n",
       "47                           X1006_splice     1           1   \n",
       "48                           X1007_splice     1           1   \n",
       "49                           X1008_splice     1           1   \n",
       "50                           X1009_splice     1           1   \n",
       "51                            X434_splice     1           1   \n",
       "52                            X475_splice     1           1   \n",
       "53                            X582_splice     1           1   \n",
       "54                            X963_splice     1           1   \n",
       "55                                p61BRAF     1           1   \n",
       "\n",
       "   Gene_most_frequent  Gene_most_frequent_count  Gene_p_unique  \\\n",
       "0               HLA-A                         1       1.000000   \n",
       "1                 MET                         1       1.000000   \n",
       "2                 MET                         1       1.000000   \n",
       "3              EIF1AX                         1       1.000000   \n",
       "4                  AR                         1       1.000000   \n",
       "5                  AR                         1       1.000000   \n",
       "6               KMT2A                         1       1.000000   \n",
       "7               EP300                         1       1.000000   \n",
       "8               CASP8                         1       1.000000   \n",
       "9               FBXW7                         1       1.000000   \n",
       "10               TP53                         1       1.000000   \n",
       "11             DNMT3B                         1       1.000000   \n",
       "12                KIT                        34       0.557789   \n",
       "13               EGFR                         6       0.562500   \n",
       "14               EGFR                         1       1.000000   \n",
       "15               EGFR                         1       1.000000   \n",
       "16               EGFR                         1       1.000000   \n",
       "17               EGFR                         1       1.000000   \n",
       "18               EGFR                         1       1.000000   \n",
       "19              MED12                         1       1.000000   \n",
       "20                KIT                         1       1.000000   \n",
       "21              MED12                         1       1.000000   \n",
       "22                KIT                         1       1.000000   \n",
       "23             PDGFRB                        19       0.260274   \n",
       "24               EGFR                        15       0.230769   \n",
       "25               FLT3                        15       0.220000   \n",
       "26             CARD11                         1       1.000000   \n",
       "27             CARD11                         1       1.000000   \n",
       "28                MYC                         1       1.000000   \n",
       "29              BRCA1                         3       0.933333   \n",
       "30              BRCA1                        24       0.508704   \n",
       "31              BRCA1                        11       0.633803   \n",
       "32               PTEN                        12       0.641921   \n",
       "33              BRCA1                       157       0.266117   \n",
       "34              BRCA1                        41       0.553279   \n",
       "35              CCNE1                         1       1.000000   \n",
       "36               PTEN                         5       0.822222   \n",
       "37               TP53                        50       0.452261   \n",
       "38               TP53                        15       0.691057   \n",
       "39               TERT                         1       1.000000   \n",
       "40              EP300                         1       1.000000   \n",
       "41            BCL2L11                         1       1.000000   \n",
       "42               MSH2                        12       0.618421   \n",
       "43             TGFBR1                         1       1.000000   \n",
       "44              NTRK1                         1       1.000000   \n",
       "45              CCND1                         2       0.974576   \n",
       "46               KRAS                         1       1.000000   \n",
       "47                MET                         1       1.000000   \n",
       "48                MET                         1       1.000000   \n",
       "49                MET                         1       1.000000   \n",
       "50                MET                         1       1.000000   \n",
       "51             PIK3R1                         1       1.000000   \n",
       "52             PIK3R1                         1       1.000000   \n",
       "53             PIK3R1                         1       1.000000   \n",
       "54                MET                         1       1.000000   \n",
       "55               BRAF                         1       1.000000   \n",
       "\n",
       "    Gene_p_most_frequent  \n",
       "0               1.000000  \n",
       "1               1.000000  \n",
       "2               1.000000  \n",
       "3               1.000000  \n",
       "4               1.000000  \n",
       "5               1.000000  \n",
       "6               0.012658  \n",
       "7               1.000000  \n",
       "8               1.000000  \n",
       "9               1.000000  \n",
       "10              1.000000  \n",
       "11              1.000000  \n",
       "12              0.170854  \n",
       "13              0.375000  \n",
       "14              1.000000  \n",
       "15              1.000000  \n",
       "16              1.000000  \n",
       "17              1.000000  \n",
       "18              1.000000  \n",
       "19              1.000000  \n",
       "20              1.000000  \n",
       "21              1.000000  \n",
       "22              1.000000  \n",
       "23              0.086758  \n",
       "24              0.384615  \n",
       "25              0.300000  \n",
       "26              1.000000  \n",
       "27              1.000000  \n",
       "28              1.000000  \n",
       "29              0.050000  \n",
       "30              0.046422  \n",
       "31              0.038732  \n",
       "32              0.026201  \n",
       "33              0.039230  \n",
       "34              0.042008  \n",
       "35              0.166667  \n",
       "36              0.055556  \n",
       "37              0.035894  \n",
       "38              0.040650  \n",
       "39              1.000000  \n",
       "40              1.000000  \n",
       "41              1.000000  \n",
       "42              0.157895  \n",
       "43              1.000000  \n",
       "44              1.000000  \n",
       "45              0.016949  \n",
       "46              1.000000  \n",
       "47              1.000000  \n",
       "48              1.000000  \n",
       "49              1.000000  \n",
       "50              1.000000  \n",
       "51              1.000000  \n",
       "52              1.000000  \n",
       "53              1.000000  \n",
       "54              1.000000  \n",
       "55              1.000000  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genefreq(df, 'Variation_recoded')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## text"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### word count per entry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df.loc[:, 'Word_count']  = df[\"text\"].apply(lambda x: len(x.split()))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### top word frequencies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "condcut removal of stop words and stemming to come up with the bag of words model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "corpus = []\n",
    "for i in range(0, 3321):\n",
    "    review = re.sub('[^a-zA-Z]', ' ', df['text'][i])\n",
    "    review = review.lower()\n",
    "    review = review.split()\n",
    "    ps = PorterStemmer()\n",
    "    review = [ps.stem(word) for word in review if not word in set(stopwords.words('english'))]\n",
    "    review = ' '.join(review)\n",
    "    corpus.append(review)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "find frequent words"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "word_vectorizer = CountVectorizer(ngram_range=(1,1), analyzer='word')\n",
    "sparse_matrix = word_vectorizer.fit_transform(corpus)\n",
    "frequencies = sum(sparse_matrix).toarray()[0]\n",
    "h = pd.DataFrame(frequencies, index=word_vectorizer.get_feature_names(), columns=['frequency'])\n",
    "h.reset_index(inplace = True)\n",
    "h.sort_values('frequency',ascending = False, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "words = 'mutat, cancer, patient, protein, express, tumor, variant, kinase, domain, brca1, egfr, activation, ras, p53, exon, growth, clinical, signaling, function, pten, phospharylation, residue, resistance, raf, pathway, alk, disease, receptor, missense, breast, braf, inhibit, lung, tyrosine, acid, induce, survival, proliferation, akt, imatinib, fusion, oncogenic, transcription, deleterous, flt3, melanoma, somatic, gefitinib, brct, myc, amplification, genomic, pathogenic, benign, p21, phosphatase, sp3b1, bat3'.split(', ')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "find frequency of top words in each entry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def word_count(df, words):    \n",
    "    for word in words:\n",
    "        w =[]\n",
    "        for row in df['text']:\n",
    "            n = row.lower().count(word)\n",
    "            w.append(n)\n",
    "        df[word] = w\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged_text = word_count(df, words)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "find frequency of top words in each class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#load the training dataset\n",
    "training_variants = pd.read_csv('training_variants.csv')\n",
    "training_text = pd.read_csv('training_text.csv',sep='\\|\\|', skiprows=1, engine='python',names=[\"ID\",\"text\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#combine the two training datasets together\n",
    "merged_variants_text = pd.merge(training_variants, training_text, on = ['ID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#group the merged dataset by class and count\n",
    "text_group_count = merged_variants_text.groupby(\"Class\")['ID'].count()\n",
    "text_group_count.sort_values(ascending=False, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# check the class count results\n",
    "text_group_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# create a list that contains all the text classfied in Class 1 to 9\n",
    "class_txt_assembly = []\n",
    "for clas in range(1,10):\n",
    "    class_txt= merged_variants_text.loc[merged_variants_text['Class']==clas]['text']\n",
    "    class_txt_assembly.append(class_txt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# define a function to calculate frequency of words to filter keywords\n",
    "def classCorpus(corpus):\n",
    "    word_vectorizer = CountVectorizer(ngram_range=(1,1), analyzer='word')\n",
    "    sparse_matrix = word_vectorizer.fit_transform(corpus)\n",
    "    frequencies = sum(sparse_matrix).toarray()[0]\n",
    "    df=pd.DataFrame(frequencies, index=word_vectorizer.get_feature_names(), columns=['frequency'])\n",
    "    df.sort_values('frequency', ascending = False, inplace = True)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# get the result of frequency of words in each class\n",
    "result=[]\n",
    "for corpus in class_txt_assembly:\n",
    "    result.append(classCorpus(corpus))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# check the result of class 1 to 9\n",
    "result[0][0:200]\n",
    "result[1][0:200]\n",
    "result[2][0:200]\n",
    "result[3][0:200]\n",
    "result[4][0:200]\n",
    "result[5][0:200]\n",
    "result[6][0:200]\n",
    "result[7][0:200]\n",
    "result[8][0:200]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
